code/datums/gamemode/objectives/objective.dm

```diff
@@ -30,15 +34,17 @@
 /datum/objective_holder/proc/FindObjective(var/datum/objective/O)
 	return locate(O) in objectives
 
-/datum/objective_holder/proc/GetObjectiveString(var/check_success = 0)
+/datum/objective_holder/proc/GetObjectiveString(var/check_success = 0,var/admin_edit = 0)
 	var/dat = ""
-	if(objectives.len == 0)
-		dat += "EMPTY<br>"
-	else
+	if(objectives.len)
 		var/obj_count = 1
 		for(var/datum/objective/O in objectives)
-			dat += {"<BR><B>Objective #[obj_count++]</b>: [O.explanation_text]"}
+			var/current_completion = O.IsFulfilled()
+			dat += {"<b>Objective #[obj_count++]</b>: [O.explanation_text]
+				[admin_edit ? " - <a href='?src=\ref[src];obj_edit=\ref[O]'>(edit)</a> - <a href='?src=\ref[src];obj_delete=\ref[O]'>(remove)</a> - <a href='?src=\ref[src];obj_completed=\ref[O]'>(toggle:[current_completion ? "<font color='green'>SUCCESS" : "<font color='red'>FAILURE" ]</font>)</a>" : ""]
+				<br>"}
 			if(check_success)
-				dat += {"<BR>[O.IsFulfilled() ? "Success" : "Failed"]"}
-
-	return dat
\ No newline at end of file
+				dat += {"<BR>[current_completion ? "Success" : "Failed"]"}
+	if(admin_edit)
+		dat += "<a href='?src=\ref[src];obj_add=1'>(add personnal objective)</a>"
```

Comment:
>personnal

Reply:
pshh... nothing head of personnal kid

Reply:
cut me some slack, we say "personnel" in french >__>

Reply:
Ouais ce n'est pas un grand problème